src/core/server/legacy/integration_tests/logging.test.ts

```diff
@@ -0,0 +1,238 @@
+        await root.start();
+      }, 30000);
+
+      afterAll(async () => {
+        await root.shutdown();
+      });
+
+      beforeEach(() => {
+        mockConsoleLog.mockClear();
+        mockStdout.mockClear();
+      });
+
+      it('when context matches', async () => {
+        root.logger.get('test-file').info('handled by NP');
+
+        expect(mockConsoleLog).toHaveBeenCalledTimes(1);
+        const loggedString = getPlatformLogsFromMock(mockConsoleLog);
+        expect(loggedString).toMatchInlineSnapshot(`
+          Array [
+            "[xxxx-xx-xxTxx:xx:xx.xxxZ][INFO ][test-file] handled by NP",
+          ]
+        `);
+      });
+
+      it('falls back to the root legacy logger otherwise', async () => {
+        root.logger.get('test-file-legacy').info('handled by LP');
+
+        expect(mockStdout).toHaveBeenCalledTimes(1);
+
+        const loggedString = getLegacyPlatformLogsFromMock(mockStdout);
+        expect(loggedString).toMatchInlineSnapshot(`
+          Array [
+            "  log   [xx:xx:xx.xxx] [info][test-file-legacy] handled by LP
+          ",
+          ]
+        `);
+      });
+    });
+
+    describe('logging config respects legacy logging settings', () => {
+      let root: ReturnType<typeof createRoot>;
+
+      afterEach(async () => {
+        mockConsoleLog.mockClear();
+        mockStdout.mockClear();
+        await root.shutdown();
+      });
+
+      it('"silent": true', async () => {
+        root = createRoot({ silent: true });
+
+        await root.setup();
+        await root.start();
+
+        const platformLogger = root.logger.get('test-file');
+        platformLogger.info('info');
+        platformLogger.warn('warn');
+        platformLogger.error('error');
+
+        expect(mockConsoleLog).toHaveBeenCalledTimes(3);
+
+        expect(getPlatformLogsFromMock(mockConsoleLog)).toMatchInlineSnapshot(`
+          Array [
+            "[xxxx-xx-xxTxx:xx:xx.xxxZ][INFO ][test-file] info",
+            "[xxxx-xx-xxTxx:xx:xx.xxxZ][WARN ][test-file] warn",
+            "[xxxx-xx-xxTxx:xx:xx.xxxZ][ERROR][test-file] error",
+          ]
+        `);
+
+        mockStdout.mockClear();
+
+        const legacyPlatformLogger = root.logger.get('test-file-legacy');
+        legacyPlatformLogger.info('info');
+        legacyPlatformLogger.warn('warn');
+        legacyPlatformLogger.error('error');
+
+        expect(mockStdout).toHaveBeenCalledTimes(0);
+      });
+
+      it('"quiet": true', async () => {
+        root = createRoot({ quiet: true });
+
+        await root.setup();
+        await root.start();
+
+        const platformLogger = root.logger.get('test-file');
+        platformLogger.info('info');
+        platformLogger.warn('warn');
+        platformLogger.error('error');
+
+        expect(mockConsoleLog).toHaveBeenCalledTimes(3);
+
+        expect(getPlatformLogsFromMock(mockConsoleLog)).toMatchInlineSnapshot(`
+          Array [
+            "[xxxx-xx-xxTxx:xx:xx.xxxZ][INFO ][test-file] info",
```

Comment:
We can stripe timestamps. I decided to keep it to see the difference in the format until we have separate tests for them.